script/cache_critical_dns

```diff
@@ -278,7 +301,7 @@ def run(hostname_vars)
     name = ENV[var]
     HOST_RESOLVER_CACHE[var] ||= ResolverCache.new(
       if (srv_name = env_srv_name(var))
-        SRVName.new(srv_name)
+        SRVName.new(srv_name, env_srv_var(var))
```

Comment:
This is probably a matter of subjective preference, but I would pass an instance of `PrioFilter` here instead of an environment variable name.  This would decouple `SRVName` from the global and any notion of environment variables.  All it really needs is a value on which it can invoke `within_threshold?`

Reply:
Agreed, passing in the filter directly is much nicer.